server/src/main/java/com/flipkart/varadhi/cluster/VaradhiClusterManager.java

```diff
@@ -9,15 +11,15 @@
  * node resources, local ip & hostname, so that they can be used while registering to the cluster and other nodes can
  * be notified of the same.
  */
-public interface ClusterManager {
+public interface VaradhiClusterManager {
 
-    List<NodeInfo> getAllMembers();
+    Future<List<MemberInfo>> getAllMembers();
 
     void addMembershipListener(MembershipListener listener);
 
-    MessageChannel connect(String nodeId);
+    MessageRouter getRouter(Vertx vertx);
 
-    // TODO: Any publish to all methods?
+    MessageExchange getExchange(Vertx vertx);
```

Comment:
any reason to call it exchange?

from first reading, the difference between exchange and router is not that clear.
I see that exchange & router are apposite of each other; similar to producer vs consume.

Reply:
Open for better names here.. 
Exchange vs Router is Produce vs Consume as you mentioned.